pages/home.page.ts

```diff
@@ -0,0 +1,32 @@
+import { Locator, Page } from "@playwright/test"
+import { commonData } from "../data/common.data";
+import { homePageSelectors } from "../selectors/homePage.selectors";
+import { profilePageSelectors } from "../selectors/profilePage.selectors";
+import { Util } from "../utils/util";
+import { base } from "./base.page";
+
+export class homePage implements base{
+    page: Page;
+    util: Util;
+    navigationMenu: Locator;
+    viewProfileMenuButton: Locator;
+    yourProfileDropDownMenuItem: Locator;
+
+    constructor(page: Page){
+        this.page = page;
+        this.util = new Util(page);
+        this.navigationMenu = this.page.locator(homePageSelectors.navigationMenu);
+        this.viewProfileMenuButton = this.page.locator(homePageSelectors.viewProfileMenuButton);
+        this.yourProfileDropDownMenuItem = this.page.locator(homePageSelectors.yourProfileDropDownMenuItem);

```

Comment:
we specifically asked not to use any locator inside the constructor, in addition, locators are sent to the page functions as it might change or need to use a different one.

if its not the case (static locator), please move any locator to the right function